cpp/ql/lib/semmle/code/cpp/models/implementations/StructuredExceptionHandling.qll

```diff
@@ -7,3 +7,25 @@ class WindowsDriverFunction extends ThrowingFunction {
 
   final override predicate mayThrowException(boolean unconditional) { unconditional = true }
 }
+
+/**
+ * Calls to functions in a try statement with no body
+ * are assumed to potentially (not unconditionally) throw anything.
+ */
+class UndefinedFunction extends ThrowingFunction {
+  UndefinedFunction() {
+    exists(Stmt t, Call c |
+      (t instanceof TryStmt or t instanceof MicrosoftTryStmt) and
+      t.getChild(0).(Stmt).getAChild*() = c and
```

Comment:
We don't really like using the raw child relation directly since that makes it really hard to add children in the future. How about this version instead (which also removes the need for casting to `Stmt`).
```suggestion
      t.(TryStmt).getStmt().getAChild*() = c or
      t.(MicrosoftTryStmt).getStmt().getAChild*() = c
```